docs/source/torch.rst

```diff
@@ -140,6 +140,42 @@ Indexing, Slicing, Joining, Mutating Ops
     vstack
     where
 
+.. _accelerators:
+
+Accelerators
+----------------------------------
+An accelerator is a specialized device designed to significantly enhace performance
+through parallel computation. It empolys a Stream/Event system to synchronize and
+control asynchronous operations effectively. Every accelerator should incorporate
+the concepts of Device, Stream, and Event. As of today, accelerator devices
+are (in no particular order) :doc:`"CUDA" <cuda>`, :doc:`"MTIA" <mtia>`, :doc:`"XPU" <xpu>`, and PrivateUse1.
+
+Device
+~~~~~~~~~~~~~~~~~~~~~~~~
+A host system can have multiple Devices of the same type. Stream and Events are in
+association with Device. It allows the switch of the specific device to the current device,
+which means that the subsequent operations (device memory allocated or kernel launched)
+from this host thread will reside or execute on the specific device.
+
+Stream
+~~~~~~~~~~~~~~~~~~~~~~~~
+An in-order queue of executing the respective tasks asynchronously in first in first out (FIFO) order.
+It can control or synchronize the execution of other Streams or block the current host thread to ensure
+the correct task sequencing.
+
+Event
+~~~~~~~~~~~~~~~~~~~~~~~~
+Query and record Stream status to identify or control dependencies across Stream and measure timing.
+
+.. _default_accelerator:
+
+Default Accelerator
```

Comment:
I think a single sentence above is enough for this, WDYT?

Reply:
OK to me.